include/wil/resource.h

```diff
@@ -2528,6 +2528,19 @@ namespace wil
 
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
     typedef unique_any<PSID, decltype(&::FreeSid), ::FreeSid> unique_sid;
+    typedef unique_any_handle_null_only<decltype(&::DeleteBoundaryDescriptor), ::DeleteBoundaryDescriptor> unique_boundary_descriptor;
+
+    namespace details
+    {
+        template<ULONG flags>
+        inline void __stdcall ClosePrivateNamespaceHelper(HANDLE h) WI_NOEXCEPT
+        {
+            ::ClosePrivateNamespace(h, flags);
+        }
+    }
+
+    template <ULONG flags = 0>
+    using unique_private_namespace = unique_any_handle_null_only<decltype(details::ClosePrivateNamespaceHelper<flags>), &details::ClosePrivateNamespaceHelper<flags>>;

```

Comment:
The usage of this is a bit odd. If you want the "default", you'd need to declare a variable of the type `unique_private_namespace<>`. Since it appears as though there's only two "useful" calls, it might make sense to have two type alias - e.g. `unique_private_namespace_close` and `unique_private_namespace_destroy` - and then get rid of the template argument.

Reply:
I thought about it, and decided to make it parametrized in case more flags are added in the future. I can change it to `unique_private_namespace_close` and `unique_private_namespace_destroy` if you prefer.

Reply:
Idk how likely that would be. Future proofing at the expense of usability is generally not the greatest option. It's probably fine to keep the parameterized alias, but it still seems beneficial to have the two other well-named aliases.

Reply:
Added.